### PR TITLE
Adding parse_expr in the sympify.py docstring 

### LINKED ISSUE #21207: Document parse_expr in the sympify() docstring
In the example below, sympy raises a fuss because it is missing a multiplier between `2` and `x`.
It looks like it attempts to sympify the expression `'could not parse '2x'`, which is strange and confusing.

```python
import sympy as sp
sp.sympify("2x+1")

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
ValueError: Error from parse_expr with transformed code: "Integer (2 )Symbol ('x' )"

The above exception was the direct cause of the following exception:

SyntaxError                               Traceback (most recent call last)
c:\users\thomasaar\documents\github\sympy\sympy\core\sympify.py in sympify(a, locals, convert_xor, strict, rational, evaluate)
    478         a = a.replace('\n', '')
--> 479         expr = parse_expr(a, local_dict=locals, transformations=transformations, evaluate=evaluate)
    480     except (TokenError, SyntaxError) as exc:

c:\users\thomasaar\documents\github\sympy\sympy\parsing\sympy_parser.py in parse_expr(s, local_dict, transformations, global_dict, evaluate)
   1016     except Exception as e:
-> 1017         raise e from ValueError(f"Error from parse_expr with transformed code: {code!r}")
   1018 

c:\users\thomasaar\documents\github\sympy\sympy\parsing\sympy_parser.py in parse_expr(s, local_dict, transformations, global_dict, evaluate)
   1014     try:
-> 1015         return eval_expr(code, local_dict, global_dict)
   1016     except Exception as e:

c:\users\thomasaar\documents\github\sympy\sympy\parsing\sympy_parser.py in eval_expr(code, local_dict, global_dict)
    908     """
--> 909     expr = eval(
    910         code, global_dict, local_dict)  # take local objects in preference

SyntaxError: invalid syntax (<string>, line 1)

During handling of the above exception, another exception occurred:

SympifyError                              Traceback (most recent call last)
<ipython-input-37-d8899059adbf> in <module>
      1 import sympy as sp
----> 2 sp.sympify("2x")

c:\users\thomasaar\documents\github\sympy\sympy\core\sympify.py in sympify(a, locals, convert_xor, strict, rational, evaluate)
    479         expr = parse_expr(a, local_dict=locals, transformations=transformations, evaluate=evaluate)
    480     except (TokenError, SyntaxError) as exc:
--> 481         raise SympifyError('could not parse %r' % a, exc)
    482 
    483     return expr

SympifyError: Sympify of expression 'could not parse '2x'' failed, because of exception being raised:
SyntaxError: invalid syntax (<string>, line 1)
```

### PR DESCRIPTION
**Pull request by cjaitej**
#### References to other Issues or PRs

- Fixes #21207
- Reference #21230
This PR adds documentation of parse_expr in the sympify.py docstring.

#### Brief description of what is fixed or changed
"sympify" function parse the input if it follows the python-syntax. In order to parse a non-python syntax use "parse_exp".

NO ENTRY

### CODE REVIEW

File: sympy/core/sympify.py [hunk 1]
```diff
@@ -138,6 +138,16 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
...
     SympifyError: SympifyError: "could not parse 'x***2'"
+    >>> sympify("2x")
+    Traceback (most recent call last):
+    ...
+    SympifyError: SympifyError: "could not parse '2x' "
+
+    sympify function parse the input if it follows the python-syntax.
+    To parse a non-python syntax use "parse_exp":
+    >>> from sympy.parsing.sympy_parser import parse_expr
+    >>> parse_expr("2x", transformations="all")
```
Comment (oscarbenjamin) [inline sympy/core/sympify.py:149]: This needs to be worded better.

File: sympy/core/sympify.py [hunk 2]
```diff
@@ -338,6 +348,10 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
         their SymPy equivalents. If True the expression will be evaluated
         and the result will be returned.
 
+    See Also
+    ========
+    parse_expr
```
Comment (oscarbenjamin) [inline sympy/core/sympify.py:353]: This is a good change

File: sympy/core/sympify.py (hunk 1 above)
Comment (cjaitej) [reply sympy/core/sympify.py:149]: I will make the necessary changes.

**Comment by cjaitej** [comment]
@oscarbenjamin 
My PR is failing the tests due to error:
AssertionError: File contains trailing whitespace: /home/runner/work/sympy/sympy/sympy/core/sympify.py, line 149.

I did check line 149 and removed all the whitespace, but I am still facing the same error.

**Comment by ThePauliPrinciple** [comment]
There is a trailing whitespace after
`If the expression is not following python syntax, use "parse_exp" to parse `

File: sympy/core/sympify.py [hunk 3]
```diff
@@ -138,6 +138,17 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     Traceback (most recent call last):
     ...
     SympifyError: SympifyError: "could not parse 'x***2'"
+    >>> sympify("2x")
+    Traceback (most recent call last):
+    ...
+    SympifyError: SympifyError: "could not parse '2x' "
+    
+    Sympification happens, if the expression follows the python syntax.
+    If the expression is not following python syntax, use "parse_exp" to parse
```
Comment (ThePauliPrinciple) [inline sympy/core/sympify.py:147]: trailing whitespace

File: sympy/core/sympify.py [hunk 4]
```diff
@@ -138,6 +138,17 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     Traceback (most recent call last):
     ...
     SympifyError: SympifyError: "could not parse 'x***2'"
+    >>> sympify("2x")
+    Traceback (most recent call last):
+    ...
+    SympifyError: SympifyError: "could not parse '2x' "
+
```
Comment (ThePauliPrinciple) [inline sympy/core/sympify.py:145]: Maybe it is the tab on this line?

Comment (cjaitej) [reply sympy/core/sympify.py:145]: Test says, Problem is in line 149.

File: sympy/core/sympify.py (hunk 1 above)
Comment (cjaitej) [reply sympy/core/sympify.py:150]: @oscarbenjamin 
I made changes to the documentation, Can you review this again?  : )

**Pull request closed without merging**